docs/developer-guide/typeConversions.md

```diff
@@ -0,0 +1,211 @@
+    HELICS_DATA_TYPE_COMPLEX = 3,
+    /** an array of doubles*/
+    HELICS_DATA_TYPE_VECTOR = 4,
+    /** a complex vector object*/
+    HELICS_DATA_TYPE_COMPLEX_VECTOR = 5,
+    /** a named point consisting of a string and a double*/
+    HELICS_DATA_TYPE_NAMED_POINT = 6,
+    /** a boolean data type*/
+    HELICS_DATA_TYPE_BOOLEAN = 7,
+    /** time data type*/
+    HELICS_DATA_TYPE_TIME = 8,
+    /** raw data type*/
+    HELICS_DATA_TYPE_RAW = 25,
+    /** type converts to a valid json string*/
+    HELICS_DATA_TYPE_JSON = 30,
+    /** the data type can change*/
+    HELICS_DATA_TYPE_MULTI = 33,
+    /** open type that can be anything*/
+    HELICS_DATA_TYPE_ANY = 25262
+```
+
+When this data is received it can be received as any of the available types, thus there are no restrictions on which function is used based on the data that was sent. That being said not all conversions are lossless. The following image shows which conversions are lossless through round trip operations. On the sending side the same is also true in that in any of the setValue methods may be used regardless of which the actual transmission data type is set to.
+
+![Lossless type conversion](../img/LossLessConversion_chart.svg)
+
+For the remaining conversions the loss it typically numerical or comes with conditions. For example Converting a vector to a complex number you can think of a complex number as a two element vector. Thus for 1 or 2 element vectors the conversion will be lossless. For 3 or more element vectors the remaining elements are discarded. For conversion of any vector to a single value, double or integral the value is collapsed using the vector norm. Thus for single element vectors or real valued complex numbers the result is equivalent for others there is a reductions in information. Which may be desired in some cases. Bool values get reduced to `false` is the numerical value is anything but `0` and true otherwise.
+
+Conversion from strings to numeric values assume the string encodes a number otherwise it results in an invalid value.
+
+### Invalid Values
+
+Invalid values are returned if there is no value or the conversion is invalid for some reason, typically string to numeric value failures, in most cases an empty value equivalent is returned.
+
+- INT -> -9,223,372,036,854,775,808 ( `numeric_limits<std::int64_t>::min()`)
+- DOUBLE -> `-1e49`
+- COMPLEX -> {-1e49,0.0}
+- VECTOR -> {}
+- COMPLEX_VECTOR -> {}
+- STRING -> ""
+- NAMED_POINT -> {"", nan}
+- BOOL -> false
+- TIME -> Time::minVal() = `Time(numeric_limits<std::int64_t>::min())`
+
+## How types are used in HELICS
+
+### Interface specification
+
+For Value based interfaces types are used in a number of locations. For publications the most important part is specifying the type of the publication. This determines the type of data that gets transmitted. It can be a specific known type, a custom type string that is user defined or an `any` type.
```

Comment:
```suggestion
For Value based interfaces types are used in a number of locations. For publications the most important part is specifying the type of the publication. This determines the type of data that gets transmitted. It can be a specific known type, a custom type string that is user defined, or an `any` type.
```